Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Adding daemontools-run entry to /etc/inittab if required.

diff -Naurp lxc.orig/templates/lxc-debian.in lxc/templates/lxc-debian.in
--- lxc.orig/templates/lxc-debian.in	2011-10-04 14:00:52.892682395 +0200
+++ lxc/templates/lxc-debian.in	2011-10-04 14:04:08.165650118 +0200
@@ -56,6 +56,15 @@ c3:12345:respawn:/sbin/getty 38400 tty3
 c4:12345:respawn:/sbin/getty 38400 tty4 linux
 EOF
 
+    # add daemontools-run entry
+    if [ -e $rootfs/var/lib/dpkg/info/daemontools.list ]; then
+        cat <<EOF >> $rootfs/etc/inittab
+#-- daemontools-run begin
+SV:123456:respawn:/usr/bin/svscanboot
+#-- daemontools-run end
+EOF
+    fi
+
     # disable selinux in debian
     mkdir -p $rootfs/selinux
     echo 0 > $rootfs/selinux/enforce
